### DISCUSSION TITLE
Cable Launch

### DISCUSSION THREAD

**Discussion by spacek531** [OpenRCT2/OpenRCT2#15745] [Feature requests]
### Describe the feature

I think a cable launch would be cool and good.

Hydraulic cable launches produced by Intamin would fit on Giga coaster. Pneumatic launches produced by S&S would fit on LIM-Launched coaster. Flywheel and gravity launches produced by Schwarzkopf would fit on Looping coaster.

Intamin and S&S launches have retracting eddy current retarders, which I would like to include. Another flag can be co-opted to differentiate regular cable lift hill and cable launch with retracting fins, or a brand new TrackElemType can be introduced to make the code simpler. A new cable launch terminus piece would have to be added.

This would slot in well with the ability for cable lift hills to pick up from block brakes.

How I think it would work:
Block brake is closed. Fins extended. Cable car reverses from its terminus into the ready position just outside the block brake. When the train is waiting on the block and the cable car is at the ready position, the cable car reverses and picks up the train (standard cable lift hill behavior).

Once the train is picked up, the train waits for the fins to retract all as one. The cable car and train launch. As the tail of the train passes each track piece, the fins on that track piece extend.

If the train rolls back, the fins slow the train down to a manageable speed like the LSM freefall coaster. A counter in the launch can keep track of how many rollbacks have occurred and increase the power slightly each consecutive launch until the train completes the cycle.

Giga and LIM-Launched would have retractable fin sprites, while Looping Coaster would not.

### Feature type

Other

**Comment by ThiaJay** [comment]
This is how Thunder Looper at Alton Towers worked isn't it?

**Comment by X123M3-256** [comment]
I have a working prototype https://i.imgur.com/50NYoWm.gif. This implementation allows the launch to start from a block brake and also allows the same for the cable lift; no dedicated launch start piece.

There is no functional handling of rollbacks, the really hard part with that would be getting the launch block to not clear until the train is past the highest point of the ride (otherwise, it won't be able to reset because another train will have moved into position on the launch block).

I'm not sure what should be done about the looping coaster since that does not have anti-rollbacks by design, it's a shuttle coaster. Right now mine are just cosmetic so it's as simple as not providing sprites, but I'd like them to be functional. I am thinking that these might have to be different pieces. The other issue is that a Schwarzkopf catch car technically engages the car at the rear, which my implementation doesn't allow for.

**Reply by TheAtomTwister** [reply]
I'll bet there is a piece that you could make that acts like block brakes... or perhaps a piece trait trait that could be added to ride pieces to mark block sections.  This would cause the train to have to go past a certain point before the next train enters the launch section when operating in block sectioned mode.
